code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/dao/AuthDao.java

```diff
@@ -18,7 +22,29 @@
 
     Account createAccount(String name, String kind, String externalId, String externalType);
 
+    Account createProject(String name, String description);
+
+    Account createDefaultProject(Account account);
+
     void updateAccount(Account account, String name, String kind, String externalId, String externalType);
 
-    List<? extends Account> getAccessibleProjects(String userAccountId, List<String> orgIds, List<String> teamIds);
+    List<Account> getAccessibleProjects(Set<ExternalId> externalIdSet, boolean isAdmin, Long policyAccount);
+
+    boolean hasAccessToProject(long projectId , Long policyAccount, boolean isAdmin, Set<ExternalId> externalIdSet);
+
+    boolean isProjectOwner(long projectId , Long policyAccount, boolean isAdmin, Set<ExternalId> externalIdSet);
+
+    boolean isProjectMember(long projectId , Long policyAccount, boolean isAdmin, Set<ExternalId> externalIdSet);
+
+    Account updateProject(Account project, String name, String description);
+
+    List<? extends ProjectMember> getActiveProjectMembers(long projectId);
+
+    ProjectMember getProjectMember(long id);
+
+    boolean hasAccessToAProject(Set<ExternalId> externalIds, boolean isAdmin, Long policyAccount);
```

Comment:
James, can we rename this method? As obviously we don't check that the account has an access to any specific project